x-pack/plugins/enterprise_search/public/applications/workplace_search/views/oauth_authorize/oauth_authorize_logic.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { JSON_HEADER as headers } from '../../../../../common/constants';
+import { clearFlashMessages, flashAPIErrors } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { parseQueryParams } from '../../../shared/query_params';
+
+interface OAuthPreAuthServerProps {
+  client_id: string;
+  client_name: string;
+  redirect_uri: string;
+  response_type: string;
+  scope: string;
+  state: string;
+  status: string;
+}
+
+export interface OAuthPreAuthorization {
+  clientId: string;
+  clientName: string;
+  redirectUri: string;
+  responseType: string;
+  rawScopes: string;
+  scopes: string[];
+  state: string;
+}
+
+interface OAuthAuthorizeValues {
+  dataLoading: boolean;
+  buttonLoading: boolean;
+  cachedPreAuth: OAuthPreAuthorization;
+}
+
+interface OAuthAuthorizeActions {
+  setServerProps(serverProps: OAuthPreAuthServerProps): OAuthPreAuthServerProps;
+  initializeOAuthPreAuth(queryString: string): string;
+  allowOAuthAuthorization(): void;
+  denyOAuthAuthorization(): void;
+  setButtonNotLoading(): void;
+}
+
+const route = '/api/workplace_search/oauth/authorize';
+
+export const OAuthAuthorizeLogic = kea<MakeLogicType<OAuthAuthorizeValues, OAuthAuthorizeActions>>({
+  path: ['enterprise_search', 'workplace_search', 'oauth_authorize_logic'],
+  actions: {
+    setServerProps: (serverProps: OAuthPreAuthServerProps) => serverProps,
+    initializeOAuthPreAuth: (queryString: string) => queryString,
+    allowOAuthAuthorization: () => null,
+    denyOAuthAuthorization: () => null,
+    setButtonNotLoading: () => null,
```

Comment:
These can be simplified as just properties. Methods aren't need if they aren't returning anything
```suggestion
    allowOAuthAuthorization: null,
    denyOAuthAuthorization: null,
    setButtonNotLoading: null,
```